src/screens/profile/profile.js

```diff
@@ -0,0 +1,314 @@
+import ImagePicker from 'react-native-image-picker';
+import { profile } from '../../../assets/images';
+import { ScrollView } from 'react-native-gesture-handler';
+import { _getUser } from '../../utils/asyncUtil';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { registerUserProfile } from './ducks/ProfileScreen.actions';
+
+const { normalize, heightScale, widthScale } = scaling;
+
+class ProfileScreen extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isEdit: false,
+            avatar: profile,
+            userData: {}
+        }
+        this.multiSelect = React.createRef()
+    }
+
+    async componentDidMount() {
+        const { usersProfile } = this.props;
+        let user = await _getUser();
+        let userData = usersProfile[user];
+        console.log(usersProfile, user)
+        this.setState({ userData, avatar: userData.avatar });
+    }
+
+    onChange = (event, selectedDate) => {
+        const currentDate = selectedDate || date;
+        setShow(Platform.OS === 'ios');
+        this.setState({ date: currentDate });
+    };
+
+
+    handlePicker = () => {
+        // console.log('edit');
+        ImagePicker.showImagePicker({}, (response) => {
+            console.log('Response = ', response);
+
+            if (response.didCancel) {
+                console.log('User cancelled image picker');
+            } else if (response.error) {
+                console.log('ImagePicker Error: ', response.error);
+            } else if (response.customButton) {
+                console.log('User tapped custom button: ', response.customButton);
+            } else {
+                this.setState({ avatar: response });
+                // here we can call a API to upload image on server
+            }
+        });
+    };
+
+    /**
+     * Formik validation schema for otp
+     */
+    validationSchemaMobile = Yup.object().shape({
+        name: Yup.string().required('Name is required'),
+        dob: Yup.date()
+            .typeError('Invalid Date')
+            .required('D.O.B. is required'),
+        gender: Yup.string().required('Gender is required'),
+        interest: Yup.string().required('Interest is required'),
+        location: Yup.string().required('Location is required'),
+    });
+
+    handleSubmit = async (formData) => {
+        const { registerUserProfile } = this.props;
+        this.setState({ isEdit: false })
+        let user = await _getUser();
+        registerUserProfile(user, {...formData, avatar: this.state.avatar});
+
+    }
+
+    getInitialValues = () => {
+        const { userData = {} } = this.state;
+        const {
+            name = '',
+            dob = '',
+            gender = '',
+            interest = '',
+            location = '',
+        } = userData;
+        return {
+            name,
+            dob,
+            gender,
+            interest,
+            location,
+        };
```

Comment:
Umm, this could be optimized. 
Instead of 
 const {
            name = '',
            dob = '',
            gender = '',
            interest = '',
            location = '',
        } = userData;
        return {
            name,
            dob,
            gender,
            interest,
            location,
        };

You should use
return {
            userData.name || ' ', // similar syntax for other properties
        }